modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/spring/BeanDefinitionFactory.java

```diff
@@ -130,16 +130,18 @@
   private final BeanDefinitionCreator<CreateDslParamGroupBeanDefinitionRequest> dslParamGroupProcessor;
   private final BeanDefinitionCreator<CreateParamBeanDefinitionRequest> paramProcessor;
   private final ObjectFactoryClassRepository objectFactoryClassRepository = new ObjectFactoryClassRepository();
+  private final boolean enableByteBuddy;
```

Comment:
add a TODO to remove this flag when the feature flag is removed. Reference such TODO in the feature flag definition in mule-api as well.